Libs/RaptureCore/src/main/java/rapture/kernel/DocApiImpl.java

```diff
@@ -786,6 +792,27 @@ public void runIndex(CallingContext context, IndexScriptPair indexScriptPair, St
             if (notEmpty.contains(uri)) continue;
             deleteDoc(context, uri);
         }
+        
+        if (recurse) {
+            String auth = new RaptureURI(docUri).getAuthority();
+            int len = auth.length() + 4 + Scheme.DOCUMENT.toString().length();
```

Comment:
Here's what we are doing:
Assume that we deleted the document://a/b/c/d/e/f/g/h/i/j/k
Now go back up the chain and if the directories are empty then remove them,
This means calling new RaptureURI each time - using substring is much quicker and simpler.
However I can make the change.